spec/ics-2-consensus-requirements/README.md

```diff
@@ -0,0 +1,124 @@
+    * `v` is the verifier, proves child `Commit.p` and returns the updated `TrustedCommit`
+    * `r` is the `AccumuatorRoot`, used to prove internal state
+
+* `Commit` is a blockchain header which provides information to update `TrustedCommit`, 
+  submitted to one blockchain to update the stored `TrustedCommit`.
+  Defined as 2-tuple `(p :: CommitProof, tc :: TrustedCommit`, where
+    * `p` is the commit proof used by `TrustedCommit.v` to verify
+    * `tc` is the new `TrustedCommit` which will replace the existing one after the verification
+ 
+* `Consensus` is a blockchain consensus algorithm which generates valid `Commit`s.
+  Defined as a function `Commit -> [Transaction] -> Commit`
+
+* `Blockchain` is a subset of `(TrustedCommit, [Commit])`, generated by a `Consensus`.
+
+#### Requirements
+
+Consensus verification requires the following accumulator primitives with datastructures and
+properties as defined in ICS23:
+
+* `AccumulatorRoot`
+
+#### Subprotocols
+
+##### Verifier
+
+Verifiers prove new `Commit` generated from a blockchain. It is expected to prove 
+a `Commit` efficiently; efficient then replaying `Consensus` logic for given parent `Commit`
+and the transactions. 
+
+##### Accumulator Root
+
+`TrustedCommit` contains an accumulator root, which identifies the whole state of the corresponding
+blockchain at the point of time that the commit is generated. It is expected that the verifying 
+inclusion or exclusion of certain data in the accumulator is done efficient. See ICS23 for the details
+about the `AccumulatorRoot`s.
+
+##### CommitProof
+
+`CommitProof` is a type dependent on concrete implementations of `Commit`. `TrustedCommit.v` uses
+the information stored in `Commit.p` to check `Commit.tc`. 
+
+##### Consensus 
+
+`Consensus` is a blockchain protocol which actually generates a list of `Commit`. 
+
+1. Headers have no more than one direct child
+ 
+* Satisfied if: deterministic safety
+* Possible violation scenario: validator double signing, miner double spend
+
+2. Headers have at least one direct child
+
+* Satisfied if: liveness, lightclient verifier continuability
+* Possible violation scenario: synchronized halt, incompatible hard fork
+
+3. Headers' accumulator root are valid transition from the parents'
+
+* Satisfied if: decentralized block generation, well implemented state machine
+* Possible violation scenario: invariant break, validator cartel
+
+If a block is submitted to the chain(and optionally passed some challenge period for fraud proof), 
+then it is assured that the packet is finalized so the application logic can process it.
+If a header is proven to violate one of the properties, the tracking chain should detect and take
+action on the event to prevent further impact. See (link for the ICS for equivocation and fraud proof) 
+for details.
+
+### Example Implementation
+
+An example blockchain `B` is run by a single operator. If a block is signed by the operator, 
```

Comment:
Delete this section? Let's just reference sub-ICS with implementations once we write those.

"Coming soon" is fine for now.

Reply:
Added a simple example where the headers are signed by a single entity and the entity can be changed.